### DISCUSSION TITLE
Error: User not found

### DISCUSSION THREAD

**Discussion by hammoaj** [Budibase/budibase#5869] [Help]
Hi - would appreciate some help. I am working on a rather non-standard self-hosted configuration. The platform I am using is a layer over OpenShift which exposes the ability to run containers within a project (an OpenShift namespace). In order to get as far as I have, I have had to decompose the `docker-compose.yml` file to manually deploy each of the required containers with the appropriate secrets. So I am currently running `app-service`, `worker-service`, `proxy-service` and `redis-service`. I'm not running the _minio_ or _couchdb_ services as I'm using the platform's native Cloudant and S3-compatible Object Store (and have set the secrets up accordingly). One of the idiosyncrasies of the platform is that each container exposes the chosen internal port (4002, 4003) on port 80 externally, so the containers have to communicate with each other by calling port 80 of the appropriate container and I have modified the `proxy-service` config accordingly. In the `proxy-service` container, internal port 10000 is exposed on external port 80 on an external route and this is how I am accessing Budibase

I can get to the initial Budibase screen and get asked to create the first admin user. I can see that it gets created in the Cloudant database, but when I then try and log in with that user and password, I get a **User not found** error. When I look in the log for worker-service I see the following:

```
ERROR [1652434094565] (28 on worker-service-7dbf469dcb-fmdwf): User not found
    ForbiddenError: User not found
        at Object.throw (/app/node_modules/koa/lib/context.js:97:11)
        at /app/dist/api/controllers/global/auth.js:57:29
        at Generator.next (<anonymous>)
        at /app/dist/api/controllers/global/auth.js:8:71
        at new Promise (<anonymous>)
        at __awaiter (/app/dist/api/controllers/global/auth.js:4:12)
        at authInternal (/app/dist/api/controllers/global/auth.js:51:12)
        at /app/dist/api/controllers/global/auth.js:72:15
        at Generator.next (<anonymous>)
        at /app/dist/api/controllers/global/auth.js:8:71
```

I appreciate that this is a very non-standard implementation, but are there any suggestions as to how I can start debugging this?

Thanks

**Comment by melohagan** [comment]
HI @hammoaj 

In the docker compose file, there is a CURL command that is run when the couchdb-service is initiated. 

```
couch-init:
    image: curlimages/curl
    environment:
      PUT_CALL: "curl -u ${COUCH_DB_USER}:${COUCH_DB_PASSWORD} -X PUT couchdb-service:5984"
    depends_on:
      - couchdb-service
    command: ["sh","-c","sleep 10 && $${PUT_CALL}/_users && $${PUT_CALL}/_replicator; fg;"]
```

You could try running that against your Cloudant database before you create the first admin user.

**Comment by hammoaj** [comment]
Hi - thanks for the idea but Cloudant pre-populates those into the database (actually CouchDB 3 does as well if you configure it as a single node cluster). I've checked on Fauxton and the admin user is inserted into the database, it seems to be an issue when re-reading it to log in.

**Reply by slicktdog08** [reply]
Did you ever resolve this issue?

**Comment by hammoaj** [comment]
No.

It works if I use the internal CouchDB but not if I use Cloudant externally

On Tue, 21 Jun 2022 at 18:50, slicktdog08 ***@***.***> wrote:

> Did you ever resolve this issue?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Budibase/budibase/discussions/5869#discussioncomment-2996092>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFBISNVJKETVDKWMCAKAYR3VQH6EVANCNFSM5V22EJEA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 

Adam Hammond
***@***.***
